File: ios/App/App/StorageCapacityPlugin.swift
import Foundation
import Capacitor

@objc(StorageCapacityPlugin)
public class StorageCapacityPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "StorageCapacityPlugin"
    public let jsName = "StorageCapacity"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getFreeSpace", returnType: CAPPluginReturnPromise)
    ]

    @objc public func getFreeSpace(_ call: CAPPluginCall) {
        do {
            let home = URL(fileURLWithPath: NSHomeDirectory(), isDirectory: true)
            let values = try home.resourceValues(forKeys: [.volumeAvailableCapacityForImportantUsageKey])
            guard let availableBytes = values.volumeAvailableCapacityForImportantUsage else {
                call.unavailable("Available storage is not reported by this volume")
                return
            }
            call.resolve(["availableBytes": NSNumber(value: availableBytes)])
        } catch {
            call.reject("Unable to measure available storage", nil, error)
        }
    }
}

File: packages/native-core/android/src/main/java/hgh/asol/app/StorageCapacityPlugin.java
package hgh.asol.app;

import android.os.StatFs;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "StorageCapacity")
public class StorageCapacityPlugin extends Plugin {
    @PluginMethod
    public void getFreeSpace(PluginCall call) {
        try {
            long availableBytes = new StatFs(getContext().getFilesDir().getAbsolutePath()).getAvailableBytes();
            JSObject result = new JSObject();
            result.put("availableBytes", availableBytes);
            call.resolve(result);
        } catch (Exception error) {
            call.reject("Unable to measure available storage", error);
        }
    }
}
